### PR TITLE
✨ added CLUSTERS_KEEPER_EC2_INSTANCES_PREFIX env to facilitate testing (⚠️ devops)

### PR DESCRIPTION
**Pull request by sanderegg**
## What do these changes do?

This adds ```CLUSTERS_KEEPER_EC2_INSTANCES_PREFIX``` to facilitate testing, and maybe later adding the option to specifically name clusters in different deployments.

This allows to name external machines with a prefix, which is extremely useful when testing so to not mangle with other deploys.

## Dev Checklist

- [x] No ENV changes or I properly updated ENV ([read the instruction](https://git.speag.com/oSparc/osparc-ops-deployment-configuration/-/blob/configs/README.md?ref_type=heads#how-to-update-env-variables))
- related changes in gitlab: https://git.speag.com/oSparc/osparc-ops-deployment-configuration/-/merge_requests/300

### CODE REVIEW

**Review by matusdrobuliak66 — APPROVED** [review]

**Review by GitHK — APPROVED** [review]

File: services/clusters-keeper/src/simcore_service_clusters_keeper/core/settings.py [hunk 1]
```diff
@@ -219,6 +219,11 @@ class ApplicationSettings(BaseCustomSettings, MixinLoggingSettings):
         auto_default_from_env=True
     )
 
+    CLUSTERS_KEEPER_EC2_INSTANCES_PREFIX: str = Field(
+        ...,
+        description="optionally set a prefix to all machines created (useful for testing)",
```
Comment (pcrespov) [inline services/clusters-keeper/src/simcore_service_clusters_keeper/core/settings.py:224]: If i tis optional, i recommend that you set a default here as well

```python
EMPTY_STRING = ""

CLUSTERS_KEEPER_EC2_INSTANCES_PREFIX: str = Field( default=EMPTY_STRING)
```

Comment (pcrespov) [reply services/clusters-keeper/src/simcore_service_clusters_keeper/core/settings.py:224]: If it is also for debugging, I would recommand to have some wa to highlight that this is being used or even forbid it in some cases depending the level of tolerance you want. 

e.g. if you do not find it critical, you can at least print a warning within a validator
```python

@validator("CLUSTERS_KEEPER_EC2_INSTANCES_PREFIX")
def warn_if_set(cls, v):
     if v:
         _logger.warning("prefixx was set. should only be for testing etc etc")
     return v
```

Comment (pcrespov) [reply services/clusters-keeper/src/simcore_service_clusters_keeper/core/settings.py:224]: Add als an option to for remove whitespaces ... as a precaution . can be nasty if you accidently capture a whitspace when you concatenate strings! Look in pydantic Config options!

**Review by pcrespov — APPROVED** [review]
Left some suggestions

File: services/clusters-keeper/src/simcore_service_clusters_keeper/core/settings.py (hunk 1 above)
Comment (sanderegg) [reply services/clusters-keeper/src/simcore_service_clusters_keeper/core/settings.py:224]: spaces are ok. it is optional but we do not have optional anymore in the ops side of things. so yeah, I will remove the bad wording

**Pull request merged**